packages/backend/src/tokens/keys.ts

```diff
@@ -26,19 +26,9 @@ function getCacheValues() {
   return Object.values(cache);
 }
 
-function setInCache(jwk: JsonWebKeyWithKid, jwksCacheTtlInMs: number) {
+function setInCache(jwk: JsonWebKeyWithKid, shouldExpire = true) {
   cache[jwk.kid] = jwk;
-  lastUpdatedAt = Date.now();
-
-  if (jwksCacheTtlInMs >= 0) {
-    setTimeout(() => {
-      if (jwk) {
-        delete cache[jwk.kid];
-      } else {
-        cache = {};
-      }
-    }, jwksCacheTtlInMs);
-  }
```

Comment:
We already have a hard-coded 5 minute TTL for the values in the jwks cache. This setTimeout appears to be resulting in a scenario where the cache is cleared by the timeout here immediately after the 5 minute TTL is hit, so the cache remains empty for the remainder of the 5 minute TTL.